generator/src/swam/generator/witx/LoadTypeEmitTraverser.scala

```diff
@@ -0,0 +1,89 @@
+package swam.generator.witx
+
+import swam.witx.traverser.TypesTraverser
+import swam.witx.unresolved.{
+  AliasType,
+  ArrayType,
+  BaseWitxType,
+  BasicType,
+  EnumType,
+  FlagsType,
+  Handle,
+  Pointer,
+  StructType,
+  UnionType
+}
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-23
+  */
+class LoadTypeEmitTraverser(prev: String, types: Map[String, BaseWitxType], offset: String = "", mem: String = "")
+    extends TypesTraverser[String](types) {
+
+  def concatOffsets(offset: String, prev: String) = {
+    if (prev.isEmpty) offset
+    else s"$offset + $prev"
+  }
+
+  override val basicTypeTraverser = {
+    case (_, t: BasicType) =>
+      t match {
+        case BasicType.u8     => s"$mem.readByte(${concatOffsets(offset, prev)}).unsafeRunSync()\n"
```

Comment:
I am really not a fan of having `unsafeRunSync` in our code, because they are only there fo `IO`, and the library code should not assume it will be an `IO`, but some `F` having some typeclass (usually `Sync`). The `Memory` class has _unsafe_ versions of the read/write operations. If you really need to run without effect type (only there for some internal performance reasons), use them instead of this.